client/state/invites/reducer.js

```diff
@@ -139,4 +162,38 @@ export function requestingResend( state = {}, action ) {
 	return state;
 }
 
-export default combineReducers( { requesting, items, counts, requestingResend } );
+/**
+ * Returns the updated site invites deletion requests state after an action has been
+ * dispatched. The state reflects an object keyed by site ID, consisting of requested
+ * invite IDs to delete, with a string representing request status.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function deleting( state = {}, action ) {
+	switch ( action.type ) {
+		case INVITES_DELETE_REQUEST:
+			const inviteDeletionRequests = zipObject(
+				action.inviteIds,
+				map( action.inviteIds, () => 'requesting' )
+			);
+			return Object.assign( {}, state, { [ action.siteId ]: inviteDeletionRequests } );
+		case INVITES_DELETE_REQUEST_FAILURE:
+			const inviteDeletionFailures = zipObject(
+				action.inviteIds,
+				map( action.inviteIds, () => 'failure' )
+			);
+			return Object.assign( {}, state, { [ action.siteId ]: inviteDeletionFailures } );
+		case INVITES_DELETE_REQUEST_SUCCESS:
+			const inviteDeletionSuccesses = zipObject(
+				action.inviteIds,
+				map( action.inviteIds, () => 'success' )
+			);
+			return Object.assign( {}, state, { [ action.siteId ]: inviteDeletionSuccesses } );
+	}
```

Comment:
Though this situation is unlikely to cause issues in production, I think we should handle it correctly:

- A request to delete an invite is sent.
- Before the first request completes, another request to delete another invite is sent.
- The first request completes, and state is updated.  The `requesting` state of the second request is lost because we replace the entire `[ action.siteId ]` key with the invites from the second request.
- The second request completes, and state is updated.  The `success` state of the first request is lost.

The `requestingResend` reducer, which is very similar, already handles this situation correctly.  In 11fb91819f789a8793d5c5aca487b60032c60c74 I proved this by expanding an existing test (and took the opportunity to rename some awkwardly-named variables), and in 9e090f4349b66e9d821b6a6c71cfa9b11b251808 I added the equivalent tests for the new `deleting` reducer (currently failing).